src/mlib/macros.h

```diff
@@ -0,0 +1,17 @@
+#ifndef MLIB_MACROS_H_INCLUDED
+#define MLIB_MACROS_H_INCLUDED
+
+#include "./user-check.h"
+
+/**
+ * @brief Cross-C/C++ compatibility for a compound initializer to be treated as

```

Comment:
Rather than modifications to `mstr` headers (wide impact), can `csfle-markup.cpp` be made into a C source file instead (small impact)?

Reply:
It could be, but these changes are incidentally useful elsewhere.

Reply:
Can you elaborate? The changes seem applicable only to satisfy the `csfle-markup.cpp` executable. I am not aware of `mlib` being utilized elsewhere that demands C++-compatibility.

Reply:
Some parts of libmongocrypt are being merged into the server source tree, including this file, so it is important that it compile cleanly in a C++ compiler. The fact that it was compiling cleanly previously was due to GCC supporting compound-initializers in C++ mode, when it really shouldn't.